libpromises/loading.c

```diff
@@ -253,6 +255,44 @@ static void ShowContext(EvalContext *ctx)
     SeqDestroy(soft_contexts);
 }
 
+static void RenameMainBundle(EvalContext *ctx, GenericAgentConfig *config, Policy *policy)
+{
+    const char *const entry_point = EvalContextGetEntryPoint(ctx);
+    Seq *bundles = policy->bundles;
+    int length = SeqLength(bundles);
+    int main_counter = 0;
+    bool removed = false;
+    for (int i = 0; i < length; ++i)
+    {
+        Bundle *const bundle = SeqAt(bundles, i);
+        const char *name = bundle->name;
+        if (StringSafeEqual(name, "__main__"))
+        {
+            const char *const path = bundle->source_path;
+            char *abspath = GetAbsolutePath(path);
+            if (StringSafeEqual(abspath, entry_point))
+            {
+                Log(LOG_LEVEL_VERBOSE, "Redefining __main__ bundle from file %s to be main", abspath);
+                free(bundle->name);
+                name = bundle->name = xstrdup("main");
```

Comment:
Ugh. Why? This is ugly and there's no need to assign to ``name`` here, right?

Reply:
You'll have to be more specific, what are you saying "Why?" to?

Assigning to name isn't strictly necessary.

Do you dislike the renaming to "main"? I preferred this over other more complicated solutions. It seems to be pretty backwards compatible, and is easy to explain the behavior.

Reply:
I meant the double-assignment, I only consider it okay when initializing variables. The name ``main`` is fine with me.